### ISSUE TITLE
Export-XLSX with -ClearSheet or -Force not working on multiple worksheets

### ISSUE THREAD

**Issue by DarkLite1** [RamblingCookieMonster/PSExcel#13]
The following code works flawlessly when there is no file with the same name present and generates 2 different worksheets in one workbook:

    $Object1 = [PSCustomObject]@{
        Name       = 'Bob'
        FamilyName = 'Smith'
    }
    $Object2 = [PSCustomObject]@{
        Year   = '2015'
        Result = 'Great!'
    }
    $ExcelFile = "$env:TEMP\Demo.xlsx"
    $Object1 | Export-XLSX -Path $ExcelFile -WorksheetName 'People' -Table -AutoFit
    $Object2 | Export-XLSX -Path $ExcelFile -WorksheetName 'Results' -Table -AutoFit
    $Excel = New-Excel -Path $ExcelFile
    $Excel | Close-Excel -Save

But when adding the switch '-ClearSheet' at the end of 'Export-XLSX', to overwrite an existing file, it's throwing the following error:

    $Object1 | Export-XLSX -Path $ExcelFile -WorksheetName 'People' -Table -AutoFit -ClearSheet
    $Object2 | Export-XLSX -Path $ExcelFile -WorksheetName 'Results' -Table -AutoFit -ClearSheet
    # Error:
    Exception calling "Add" with "2" argument(s): "Tablename is not unique"

When using the switch '-Force' it deletes both worksheets and only leaves the last worksheet. In this case the one named 'Results'. And throws no errors.

    $Object1 | Export-XLSX -Path $ExcelFile -WorksheetName 'People' -Table -AutoFit -Force
    $Object2 | Export-XLSX -Path $ExcelFile -WorksheetName 'Results' -Table -AutoFit -Force

The desired behavior when using '-Force' (I think) would be to only overwrite that specific worksheet and not to touch other worksheets in the workbook. Could this be corrected?

**Comment by RamblingCookieMonster** [comment]
Hi!

Just to clarify the (programatically) expected behavior of those switches:

* Force present: If the file exists, overwrite it.
* Force absent: If the file exists, add a worksheet to it (fails if duplicate worksheet)
* ClearSheet present: If the worksheet exists, clear the contents (allows you to re-use formatting and so forth)
* ClearSheet absent: If the worksheet exists, fail

I'll take a look at the logic for tables, it looks like we didn't handle that case.

To work around this, the following might help:

```powershell
#Overwrite $ExcelFile if it exists
$Object1 | Export-XLSX -Path $ExcelFile -WorksheetName 'People' -Table -AutoFit -Force

#Add Worksheet
$Object2 | Export-XLSX -Path $ExcelFile -WorksheetName 'Results' -Table -AutoFit
```

Cheers!

**Comment by DarkLite1** [comment]
It would be great if there is a switch to forcefully overwrite a worksheet, without deleting the whole file.

Thank you F for the workaround, it's definitely usable for the time being.

**Comment by RamblingCookieMonster** [comment]
Cool, that would make sense, nice suggestion. Something along these lines?

* Force deletes existing file (implemented)
* ReplaceWorksheet would remove and replace an existing worksheet (logic / name TBD)
* ClearWorksheet would only clear the contents of the worksheet (implemented)

Thanks!

**Comment by DarkLite1** [comment]
Yes, that would be great!

The switch '-ReplaceWorksheet' should however not throw an error if the worksheet is not present, it should then just continue adding it. So you can use it all the time if you want to be sure to have the latest data in the workbook/sheet.

**Comment by RamblingCookieMonster** [comment]
Alrighty! Fixed this in c86c6bfad08379e006610566be2e43bd3e3ece45. Ended up calling it ReplaceSheet (consistent with ClearSheet).

Thanks for the idea!
